test-result-summary-client/src/Build/GitNewIssue.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import { params } from '../utils/query';
+import { Button, Tooltip, Card } from 'antd';
+import TestBreadcrumb from './TestBreadcrumb';
+import { GithubOutlined } from '@ant-design/icons';
+import { getParams } from '../utils/query';
+import renderDuration from './Duration';
+
+import './table.css';
+
+import moment from 'moment';
+const DAY_FORMAT = 'MMM DD YYYY, hh:mm a';
+
+export default class GitNewissue extends Component {
+    state = {
+        body: '',
+        title: '',
+    };
+
+    async componentDidMount() {
+        await this.updateData();
+    }
+
+    async updateData() {
+        const { testId, testName, duration, buildId, buildName,
+            buildUrl, machine, buildTimeStamp, javaVersion, testResult } = getParams(this.props.location.search);
+
+        let firstSeenFailure = null;
+        let failCount = 1;
+        let failMachineUrlBody = '';
+
+        if (testResult === 'FAILED') {
+            // get all history tests with strictly earlier timestamp
+            const fetchPreviousTests = await fetch(`/api/getHistoryPerTest?testId=${testId}&beforeTimestamp=${buildTimeStamp}&limit=100`, {
+                method: 'get'
+            });
+            const response = await fetchPreviousTests.json();
+
+            let machinesMap = {};
+            for (let i = 0; i < response.length; i++) {
+                const result = response[i].tests.testResult;
+                const machine = response[i].machine;
+                if (result === 'PASSED') {
+                    break;
+                } else {
+                    firstSeenFailure = response[i];
+                    machinesMap[machine] = machinesMap[machine] ? machinesMap[machine] + 1 : 1;
+                    failCount++;
+                }
+            }
+
+            Object.entries(machinesMap).forEach(([key, value]) => {
+                failMachineUrlBody += `The test failed on machine ${key} for ${value} times \n`;
+            });
+        }
```

Comment:
Thanks @OscarQQ 
- We need to include the current machine info in `failMachineUrlBody `.
- please remove `for` in the above msg.